File: apps/polyglot-sample/SampleClass.kt
class Calculator {
    var total: Int = 0

    fun add(value: Int): Int {
        total = total + value
        return total
    }

    fun subtract(value: Int): Int {
        total = total - value
        return total
    }
}

fun answer(): Int {
    return 42
}

fun main() {}

File: apps/polyglot-sample/sample_class.rs
struct Calculator {
    value: i64,
}

impl Calculator {
    fn new() -> Self {
        Calculator { value: 0 }
    }

    fn answer(&self) -> i64 {
        42
    }

    fn add(&mut self, x: i64) -> i64 {
        self.value = self.value + x;
        self.value
    }
}

fn answer() -> i64 {
    let calc = Calculator::new();
    calc.answer()
}

fn main() {}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_calculator_new() {
        let calc = Calculator::new();
        assert_eq!(calc.value, 0);
    }
}
